fastserde/avro-fastserde/src/main/java/com/linkedin/avro/fastserde/FastDeserializerGenerator.java

```diff
@@ -752,8 +762,9 @@ private void processArray(JVar arraySchemaVar, final String name, final Schema a
       action = FieldAction.fromValues(arraySchema.getElementType().getType(), false, EMPTY_SYMBOL);
     }
 
-    final JVar arrayVar = action.getShouldRead() ? declareValueVar(name, effectiveArrayReaderSchema, parentBody, true, false, true) : null;
-    /**
+    final boolean useLogicalType = action.getShouldRead() && logicalTypeEnabled(effectiveArrayReaderSchema.getElementType());
```

Comment:
Seems like `useLogicalType` is only ever used when `action.getShouldRead() == true`, so it might not matter to for `useLogicalType` to have `action.getShouldRead() &&`... but if you think that makes the code more defensive or something like that it could be fine to leave it as is I guess.

Reply:
Additional check for `action.getShouldRead()` is required because in some cases `effectiveArrayReaderSchema` is `null`, e.g. when reader schema is different than writer schema.
`effectiveArrayReaderSchema` is not `null` when `action.getShouldRead()` is `true`.